geometry_msgs/Point[] ref_points
geometry_msgs/Vector3[] ref_normals
float64 offset_distance
---
geometry_msgs/Point[] remain_ref_points
nav_msgs/Path candidate
geometry_msgs/Vector3[] candidate_normals
bool validity
